atuin/src/shell/atuin.zsh

```diff
@@ -12,6 +12,9 @@ autoload -U add-zsh-hook
 export ATUIN_SESSION=$(atuin uuid)
 
 _atuin_preexec() {
+    # skip history for empty calls
+    # remove space, \t, \n, and literal '\'
+    [[ -n "${${1//[[:space:]]/}//\\/}" ]] || return
```

Comment:
I appreciate why this is here, but would it be possible to implement it rust-side instead? we could then add some tests as well and ensure it doesn't happen for any shell

Reply:
Would this be the right place to introduce such a check?

https://github.com/atuinsh/atuin/blob/1a432b61ee4f52f3ff0da1c991aa5d605f700869/atuin/src/command/client/history.rs#L293